### PR TITLE
Fix more ansible 2.8.7 compability issues

### PR DESCRIPTION
**Pull request by wangxin**
### Description of PR

Summary:
Fixes # (issue)

Issues fixed in this PR:
* Change 'connection: local' to 'delegate_to: localhost' to support configuring ansible_python_interpreter for specific test servers
* Add pip_executable configuration option
* Fix datetime call in kickstart.py
* Fix tags issue of fanout deploy/config
* Remove uncesessary tags in vlantb and everflow_testbed testing
* New ansible parses inventory file as full path, fix issue of finding correct inventory file in pytest_runner.yml
* Fix issue of creating PTF host in conftest

More details of some of the fixed issues
#### Change 'connection: local' to 'delegate_to: localhost' to support configuring ansible_python_interpreter for specific test servers
This is half fix half an enhancement. With this fix, we can setup python virtualenv on test servers and configure the test server to use the python binary in virtualenv to run ansible tasks.

- fix aspect:
Ansible 2.8.7 uses SSLv3. The apt_key task in files like ansible/roles/vm_set/tasks/docker.yml requires at least python 2.7.9. Unfortunately python version on some linux distributions does not meet the requirement. To upgrade the system python, usually the linux OS need to be upgraded. If the linux OS cannot be upgrade somehow. Then this change is to fix this issue by adding python virtualenv support
- enhancement aspect:
If the test server is also used for other purpose. Then there is potential python package conflicts. Python package dependencies of other application running on the test server may conflict with the python packages required by ansible testing. Then this change is to enhance the sonic-mgmt testing.

Caveats:
* In the future, we should avoid using "connection: local". Always stick to "delegate_to: localhost".  Otherwise, ansible would try to use the ansible_python_interpreter configured for test server to run scripts on localhost.
* The only exception is the wait_for ansible module. Always use "connection: local" for the wait_for module. If use "wait_for" module with "delegate_to: localhost", the wait_for module will wait for the localhost port down/up.
References: 
https://docs.ansible.com/ansible/latest/inventory/implicit_localhost.html
https://docs.ansible.com/ansible/latest/user_guide/playbooks_delegation.html

#### Add pip_executable configuration option
This is a sister change of the above one. To support virtualenv, we also need to specify pip_executable for specific test server.

#### New ansible parses inventory file as full path, fix issue of finding correct inventory file in pytest_runner.yml
In Ansible 2.8.7, the inventory file name is parsed to absolute path in runtime. The pytest_runner.yml file needs to be updated accordingly to locate the correct inventory file.

#### Fix tags issue of fanout deploy/config
In ansible 2.8.7, tags is only inherited to child tasks when "import_tasks" is used. If use tags with "include_tasks", the tags only apply to the include_task task itself. Child tasks will not inherit the tags.
The fanout deploy/config related playbooks use the tags feature and need update accordingly to comply with new ansible behavior.

#### Fix issue of creating PTF host in conftest
The testbed_devices fixture depends on ansible code to find ptf_host from inventory. The API of ansible 2.8.7 has been changed. The testbed_devices fixture in conftest.py needs to be updated accordingly.

### Type of change

- [x] Bug fix
- [] Testbed and Framework(new/improvement)
- [] Test case(new/improvement)

### CODE REVIEW

File: ansible/roles/test/tasks/snmp/cpu.yml [hunk 1]
```diff
@@ -1,4 +1,59 @@
-- name: run test
-  include_tasks: roles/test/tasks/pytest_runner.yml
-  vars:
-    test_node: snmp/test_snmp_cpu.py
+# Test SNMP CPU Utilization
```
Comment (lguohan) [inline ansible/roles/test/tasks/snmp/cpu.yml]: please keep use pytest.

File: ansible/roles/test/tasks/snmp/interfaces.yml [hunk 2]
```diff
@@ -1,4 +1,37 @@
-- name: run test
-  include_tasks: roles/test/tasks/pytest_runner.yml
-  vars:
-    test_node: snmp/test_snmp_interfaces.py
+# Gather facts with SNMP version 2
```
Comment (lguohan) [inline ansible/roles/test/tasks/snmp/interfaces.yml]: please keep use pytest.

File: ansible/roles/test/tasks/snmp/lldp.yml [hunk 3]
```diff
@@ -1,4 +1,114 @@
-- name: run test
-  include_tasks: roles/test/tasks/pytest_runner.yml
-  vars:
-    test_node: snmp/test_snmp_lldp.py
+# Test checks for ieee802_1ab MIBs:
```
Comment (lguohan) [inline ansible/roles/test/tasks/snmp/lldp.yml]: please keep use pytest.

File: ansible/roles/test/tasks/snmp/pfc_counters.yml [hunk 4]
```diff
@@ -1,4 +1,15 @@
-- name: run test
-  include_tasks: roles/test/tasks/pytest_runner.yml
-  vars:
-    test_node: snmp/test_snmp_pfc_counters.py
```
Comment (lguohan) [inline ansible/roles/test/tasks/snmp/pfc_counters.yml]: please keep use pytest.

**Comment by lguohan** [comment]
LGTM, please revert back those snmp tests to use pytest.

File: ansible/roles/test/tasks/snmp/cpu.yml (hunk 1 above)
Comment (wangxin) [reply ansible/roles/test/tasks/snmp/cpu.yml]: Used wrong stragegy while resolving conflicts. Reverted now.

File: ansible/roles/test/tasks/snmp/interfaces.yml (hunk 2 above)
Comment (wangxin) [reply ansible/roles/test/tasks/snmp/interfaces.yml]: Reverted

File: ansible/roles/test/tasks/snmp/lldp.yml (hunk 3 above)
Comment (wangxin) [reply ansible/roles/test/tasks/snmp/lldp.yml]: Reverted

File: ansible/roles/test/tasks/snmp/pfc_counters.yml (hunk 4 above)
Comment (wangxin) [reply ansible/roles/test/tasks/snmp/pfc_counters.yml]: Reverted

**Comment by wangxin** [comment]
@msosyak Thanks for reviewing the code! I didn't add extra white spaces. The editor automatically trimmed ending white spaces while saving the files I touched.

@lguohan Thanks for pointing out the issus! I resolved conflicts using wrong strategy. The SNMP tests have been reverted to pytest now.

**Pull request merged**